### PR TITLE
SDIT-1580 integrate court appearance creation

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml [hunk 1]
```diff
@@ -80,6 +80,7 @@ generic-service:
     FEATURE_EVENT_INCENTIVES_PRISON_LEVEL_CHANGED: true
     FEATURE_EVENT_DELETEALL: false
     FEATURE_EVENT_COURT_SENTENCING_COURT_CASE_CREATED: true
+    FEATURE_EVENT_COURT_SENTENCING_COURT_APPEARANCE_CREATED: true
```
Comment (lbennett-moj) [inline helm_deploy/hmpps-prisoner-to-nomis-update/values.yaml:83]: events not defined in cloud platform yet, the names will almost certainly change

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtSentencingService.kt [hunk 2]
```diff
@@ -79,6 +84,72 @@ class CourtSentencingService(
...
+
+        val courtEventChargesToUpdate: MutableList<Pair<Charge, Long>> = mutableListOf()
+        val courtEventChargesToCreate: MutableList<Charge> = mutableListOf()
+        courtAppearance.charges.forEach { charge ->
+          courtCaseMappingService.getMappingGivenCourtChargeIdOrNull(charge.chargeUuid.toString())?.let { mapping ->
+            courtEventChargesToUpdate.add(Pair(charge, mapping.nomisCourtChargeId))
+          } ?: let {
+            courtEventChargesToCreate.add(charge)
+          }
+        }
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtSentencingService.kt:122]: idiomatic style on this section made it less readable

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtCasesToNomisIntTest.kt [hunk 3]
```diff
@@ -269,6 +273,239 @@ class CourtCasesToNomisIntTest : SqsIntegrationTestBase() {
...
+          eq("court-appearance-create-success"),
+          org.mockito.kotlin.check {
+            Assertions.assertThat(it["dpsCourtCaseId"]).isEqualTo(COURT_CASE_ID_FOR_CREATION)
+            Assertions.assertThat(it["nomisCourtCaseId"]).isEqualTo(NOMIS_COURT_CASE_ID_FOR_CREATION.toString())
+            Assertions.assertThat(it["offenderNo"]).isEqualTo(OFFENDER_NO)
+            Assertions.assertThat(it["mappingType"]).isEqualTo(CourtCaseMapping.MappingType.DPS_CREATED.toString())
+            Assertions.assertThat(it["dpsCourtAppearanceId"]).isEqualTo(DPS_COURT_APPEARANCE_ID)
+            Assertions.assertThat(it["nomisCourtAppearanceId"]).isEqualTo(NOMIS_COURT_APPEARANCE_ID.toString())
+            Assertions.assertThat(it["courtCharges"]).contains("[CourtChargeMappingDto(nomisCourtChargeId=12, dpsCourtChargeId=9996aa44-642a-484a-a967-2d17b5c9c5a1, label=null, mappingType=DPS_CREATED, whenCreated=null), CourtChargeMappingDto(nomisCourtChargeId=14, dpsCourtChargeId=1236aa44-642a-484a-a967-2d17b5c9c5a1, label=null, mappingType=DPS_CREATED, whenCreated=null)]")
+          },
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtCasesToNomisIntTest.kt:331]: will replace values with constants in this long string

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtSentencingService.kt [hunk 4]
```diff
@@ -123,17 +215,55 @@ fun CourtCase.toNomisCourtCase(): CreateCourtCaseRequest = CreateCourtCaseReques
...
+fun CourtAppearance.toNomisCourtAppearance(
+  courtEventChargesToCreate: List<OffenderChargeRequest>,
+  courtEventChargesToUpdate: List<ExistingOffenderChargeRequest>,
+): CourtAppearanceRequest {
+  return CourtAppearanceRequest(
+    // expecting time to be added to the remand and sentencing model at some point
+    eventDateTime = LocalDateTime.of(
+      this.appearanceDate,
+      LocalTime.MIDNIGHT,
+    ).toString(),
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtSentencingService.kt:234]: I think you could do `appearanceDate.atStartOfDay().toString()` here

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**